add the function writegraphml

CarloLucibello opened the pull request.

sbromberger commented on test/persistence.jl:
```diff
@@ -53,6 +53,11 @@ close(outread)
 
+@test writegraphml("test1.graphml", "g3", g3) == 1
+@test writegraphml("test1.graphml", "g4", g4) == 1
+rm("test1.graphml")
```
Could you use temp files here instead of hardcoded filenames? You can use [mktemp](http://docs.julialang.org/en/release-0.3/stdlib/file/#Base.mktemp) for this.

sbromberger commented:
Not sure why the tests are failing (`erdos_renyi` is indeed defined and exported). I'll take a look this evening.

sbromberger commented on src/LightGraphs.jl:
```diff
@@ -92,7 +92,7 @@ CombinatorialAdjacency,
 
 # persistence
-readgraph, readgraphml, readgml,
+readgraph, readgraphml, readgml, writegraphml
```
AH, found it. You need a comma after `writegraphml`.

CarloLucibello replied:
ah, ok. didn'it see yout comment. fixed

CarloLucibello commented:
Ok, I updated the test redirecting to stdout. I could not run the tests because of the erdos_renyi erro though. Please, since I'm quite new to Julia and to collaborative coding, please point me to any eventual mistake. 

Also, i added the writegexf function, since the format is quite similar to graphml. 

I'd also like to the possibility to add graph/vertex/edge properties to the export. Does this lies outside the scope of LightGraphs? Eventually, have you any suggestions for how the interface could be?

sbromberger commented:
> I'd also like to the possibility to add graph/vertex/edge properties to the export. Does this lies outside the scope of LightGraphs? Eventually, have you any suggestions for how the interface could be?

This is outside the scope of LightGraphs but could easily be done with another datastore. I'm using Redis to keep information about edges and vertices (mainly because it has a really nice Julia client and was easy to set up).

sbromberger commented:
Also, you might want to include helper methods like so:

```
writegexf(g::SimpleGraph) = writegexf(STDOUT, g)
```
(and one for `writegraphml` also)

to make it consistent with the other functions.

sbromberger commented on src/persistence.jl:
```diff
@@ -216,3 +216,100 @@ function readgml(filename::AbstractString, gname::AbstractString="")
+        xe = new_child(xg, "edge")
+        set_attribute(xe,"id","e$m")
+        set_attribute(xe,"source","n$(source(e)-1)")
+        set_attribute(xe,"target","n$(target(e)-1)")
```
I think these should be `src(e)` and `dst(e)`, respectively.

sbromberger commented on src/persistence.jl:
```diff
@@ -216,3 +216,100 @@ function readgml(filename::AbstractString, gname::AbstractString="")
+        xe = new_child(xedges, "edge")
+        set_attribute(xe,"id","$m")
+        set_attribute(xe,"source","$(source(e)-1)")
+        set_attribute(xe,"target","$(target(e)-1)")
```
See above re `src` and `dst`....

sbromberger commented:
Thank you!
